api/src/services/graphql/utils/replace-m2a.ts

```diff
@@ -0,0 +1,81 @@
+import type { Filter, NestedDeepQuery, SchemaOverview } from '@directus/types';
+
+export function filterReplaceM2A(filter_arg: Filter, collection: string, schema: SchemaOverview): any {
+	const filter: any = filter_arg;
+
+	for (const key in filter) {
+		const o2m_relation = schema.relations.find(
+			(relation) => relation.related_collection === collection && relation.meta?.one_field === key,
+		);
+
+		const m2o_relation = schema.relations.find(
+			(relation) => relation.collection === collection && relation.field === key,
+		);
+
+		const a2o_relation = schema.relations.find(
+			(relation) =>
+				relation.collection === collection &&
+				key.startsWith(relation.field + '__') &&
+				relation.meta?.one_allowed_collections?.some((collection) => collection === key.split('__')[1]),
+		);
```

Comment:
Could we utilize `getRelationInfo` here and extract the `relationType` 🤔

Reply:
Didn't went with getRelationInfo but very close to it so it's a bit cleaner now.